controllers/host/host_controller.go

```diff
@@ -2073,6 +2147,78 @@ func (r *HostReconciler) Reconcile(ctx context.Context, request ctrl.Request) (r
 		return common.RetrySystemNotReady, nil
 	}
 
+	if !r.CloudManager.GetHostUpdateRoutinesRunning() {
+		logHost.Info("Starting goroutines to receive host strategy updates from other controllers")
+		go func() {
```

Comment:
the goroutines could cause issues in the long term, we are keeping track of leaks here?

Reply:
Refactored the code to have a better lifecycle management of this goroutine, please take a look.
Although it looks like we are infinitely looping around, the code in the loop is executed only when data is sent through corresponding channel and once the code is executed the loop is blocked waiting to receive data in upcoming iterations.
Since it's considered ok to generally keep the channel open all the time, I think it's safe to that extent.

Reply:
Request a end of the goroutine threads when the main thread closed